### DISCUSSION TITLE
Using PR comments as commands

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#15221] [Suggest an Idea]
It's often been asked if we can support comments as commands, like some other bots/tools do, including some automerging ones. For example `@renovate rebase`, although I think we have a better solution for that (the checkbox). Another related request/idea here: https://github.com/renovatebot/renovate/discussions/15202#discussioncomment-2607679

There's a few challenges for this:
- We want this to work in CLI and not just in the hosted app
- Working in the CLI would mean checking every comment of every PR on every run
- We need to distinguish between old comments already acted upon, versus new ones we need to act upon
- We need to determine if the person making the comment is *authorized* to influence the bot (might require additional permissions)
- In the case of the app, `@renovate do something` would actually notify `github.com/renovate` (which we don't control) rather than the app which is `github.com/apps/renovate`. So we might want to avoid `@` altogether and instead do something like `#renovate` or even just `#rebase`

**Comment by modem7** [comment]
@rarkins - Would it be possible to have a checkbox for things like `squash and merge` so that it also deletes the branch after it's done in the meantime or does this pose the same challenges as above?

**Reply by rarkins** [reply]
It would be quite a different feature, please create a new discussion (not issue) to discuss

**Comment by nachtien** [comment]
I'd love this. Another cool feature would be if you could leave a comment and change the version it was going to update to.  I ran into an issue today where we couldn't update to the latest and greatest, but a version lower.

**Comment by ericcornelissen** [comment]
I'd love to see a `renovate ignore this minor version` functionality (just [like dependabot](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-pull-requests-for-dependency-updates#managing-dependabot-pull-requests-with-comment-commands)) to manage selected updates without having to update the configuration file.

**Reply by jamietanna** [reply]
That functionality is possible by closing the PR - Renovate will detect it as such

**Reply by ericcornelissen** [reply]
In my experience closing the PR ignores the specific version the upgrade was for, not the whole minor version range. It might be worth nothing that I deal exclusively with pinned dependencies.

If the behavior is as you described, I believe it would still be valuable to have the option to ignore only the specific version or ignore an entire major version (as in `renovate ignore this major version`).
